activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/test/java/org/activiti/cloud/services/query/events/handlers/ProcessCandidateStarterGroupAddedEventHandlerTest.java

```diff
@@ -58,6 +61,27 @@ public void handleShouldStoreNewProcessCandidateGroupUser() {
         assertThat(captor.getValue().getGroupId()).isEqualTo(event.getEntity().getGroupId());
     }
 
+    @Test
+    public void handleShouldNotStoreProcessCandidateGroupIfExists() {
+        //given
+        ProcessCandidateStarterGroupImpl candidateGroup = new ProcessCandidateStarterGroupImpl(UUID.randomUUID().toString(),
+                                                                                               UUID.randomUUID().toString());
+        CloudProcessCandidateStarterGroupAddedEventImpl event = new CloudProcessCandidateStarterGroupAddedEventImpl(candidateGroup);
+
+        //when
+        handler.handle(event);
+        when(entityManager.find(ProcessCandidateStarterGroupEntity.class,
+                                new ProcessCandidateStarterGroupId(candidateGroup.getProcessDefinitionId(), candidateGroup.getGroupId())))
+            .thenReturn(new ProcessCandidateStarterGroupEntity());
+        handler.handle(event);
+
+        //then
+        ArgumentCaptor<ProcessCandidateStarterGroupEntity> captor = ArgumentCaptor.forClass(ProcessCandidateStarterGroupEntity.class);
+        verify(entityManager, times(1)).persist(captor.capture());
+        assertThat(captor.getValue().getProcessDefinitionId()).isEqualTo(event.getEntity().getProcessDefinitionId());
+        assertThat(captor.getValue().getGroupId()).isEqualTo(event.getEntity().getGroupId());
```

Comment:
Then this bock becomes:
`verify(entityManager, never()).persist(any());`
What do you think?

Reply:
makes sense. I'll update tests